solidity/contracts/token/TBTCToken.sol

```diff
@@ -5,6 +5,37 @@ pragma solidity <0.9.0;
 import "./ERC20WithPermit.sol";
 import "./MisfundRecovery.sol";
 
+interface IReceiveApproval {
```

Comment:
In TBTC v1 this interface is named ITokenRecipient. Was there a reason why we don't want to name it the same?

Reply:
In retrospect, I think `ITokenRecipient` is too broad. It is about a specific way of receiving approval and *maybe* receiving tokens if the implementation decides for that.

Reply:
Sure, I'm okay with both, but what I liked about `ITokenRecipient` is that it tells me right away in its name that this contract will receive tokens. `IReceiveApproval(spender).receiveApproval` also looks a bit redundant.